anesthetic/utils.py

```diff
@@ -89,6 +90,151 @@ def quantile(a, q, w=None, interpolation='linear'):
+    if not np.all(np.logical_or(weights == 0, weights == 1)):
+        # compress_weights with ncompress<=0 assures weights \in 0,1
+        # Note that this must be done, we cannot handle weights != 1
+        # see this discussion for details:
+        # https://github.com/williamjameshandley/anesthetic/pull/188#issuecomment-1274980982
+        weights = compress_weights(weights, ncompress=-1)
+        if verbose:
+            print("Compressing weights to", np.sum(weights),
+                  "unit weight samples.")
+
+    indices = np.where(weights)[0]
+    x = samples[indices]
+
+    # Sample the confidence interval multiple times
+    # to get errorbars on confidence interval boundaries
+    ci_samples = []
+    for i in range(nsamples):
+        invCDF = sample_cdf(x, inverse=True)
+        if method == 'iso-pdf':
+            # Find smallest interval
+            def distance(Y, level=level):
+                return invCDF(Y+level)-invCDF(Y)
+            res = minimize_scalar(distance, bounds=(0, 1-level),
+                                  method="Bounded")
+            ci_samples.append(np.array([invCDF(res.x),
+                                        invCDF(res.x+level)]))
+        elif method == 'lower-limit':
+            # Get value from which we reach the desired level
+            ci_samples.append(invCDF(1-level))
+        elif method == 'upper-limit':
+            # Get value to which we reach the desired level
+            ci_samples.append(invCDF(level))
+        elif method == 'equal-tailed':
+            ci_samples.append(np.array([invCDF((1-level)/2),
+                                        invCDF((1+level)/2)]))
+        else:
+            raise ValueError("Method '{0:}' unknown".format(method))
+    ci_samples = np.array(ci_samples)
+    if np.shape(ci_samples) == (nsamples, ):
+        if verbose:
+            print(f"The {level:.0%} credibility interval is",
+                  "{0:.2g} +/- {1:.1g}".format(np.mean(ci_samples),
+                                               np.std(ci_samples, ddof=1)))
+        if return_covariance:
+            return np.mean(ci_samples), np.cov(ci_samples)
+        else:
+            return np.mean(ci_samples)
+    elif np.shape(ci_samples) == (nsamples, 2):
+        if verbose:
+            print(f"The {level:.0%} credibility interval is",
+                  "[{0:.2g} +/- {1:.1g}, {2:.2g} +/- {3:.1g}]".format(
+                    np.mean(ci_samples[:, 0]), np.std(ci_samples[:, 0],
+                                                      ddof=1),
+                    np.mean(ci_samples[:, 1]), np.std(ci_samples[:, 1],
+                                                      ddof=1)))
+        if return_covariance:
+            return np.mean(ci_samples, axis=0), \
+                   np.cov(ci_samples, rowvar=False)
+        else:
+            return np.mean(ci_samples, axis=0)
+    else:
+        raise ValueError('ci_samples in unrecognised shape')
```

Comment:
I don't think it _should_ ever not match, but I was cautious of maybe this could break if you come up with something crazy to pass.

Turned it into an else now.